shortfin/python/shortfin_apps/llm/components/debug_service.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+Some code for debugging service.py. Importing this should do nothing and import no additional dependencies if DEBUG_LLM_SERVICE == False
+"""
+
+
+import asyncio
+import logging
+from datetime import datetime
+from pathlib import Path
+import json
+import numpy as np
+import pandas as pd
+from typing import List
+from pprint import pformat
+
+logger = logging.getLogger(__name__)
+
+import os
+
+# Get environment variable, default to False if not set
+SHORTFIN_DEBUG_LLM_SERVICE = os.getenv(
+    "SHORTFIN_DEBUG_LLM_SERVICE", "False"
+).lower() in ("true", "yes", "1", "y")
+if SHORTFIN_DEBUG_LLM_SERVICE:
+    logger.info("DEBUG_LLM_SERVICE=True")
+    dump_id = 0
+    boot_timestamp = datetime.now().isoformat()
+    DEBUG_DATA_DIR = Path.home() / "sfdebug"
```

Comment:
This could be configurable. We might want to use a hidden directory (prefixed with `.`) or namespace similar outputs across this project to a `shark-ai` or `shortfin` (or this `sfdebug`) folder.

Reply:
I prefer using an env var & using a hardcoded debug data dir for simplicity, but i like the hidden dir idea. Might do ~/.shortfin/debug